Align processor architecture

ppcad opened the pull request.

ekneg54 commented on tests/unit/framework/rule_tree/test_rule_tree.py:
```diff
@@ -110,6 +110,12 @@ def test_get_rule_id(self, rule_dict):
+        rule3 = PreDetectorRule._create_from_dict(rule_dict)
+        assert rule_tree.get_rule_id(rule) == 0
+        assert rule_tree.get_rule_id(rule2) == 1
+        assert rule_tree.get_rule_id(rule3) == None
```
```suggestion
        assert rule_tree.get_rule_id(rule3) is None
```

ekneg54 commented on CHANGELOG.md:
```diff
@@ -4,12 +4,21 @@
 
 ### Breaking
 
+* change `pseudonyms` in pseudonymizer rules to `mapping` and change `target` in clusterer rules to `source_fields` like in the field_mapper
```
```suggestion
* `pseudonymizer` change rule config field `pseudonyms` to `mapping`
* `clusterer` change rule config field `target` to `source_fields`
* `generic_resolver` change rule config field `append_to_list` to `extend_target_list`
* `hyperscan_resolver` change rule config field `append_to_list` to `extend_target_list`
* `calculator` now adds the error tag `_calculator_missing_field_warning` to the events tag field instead of `_calculator_failure` in case of missing field in events
* `domain_label_extractor` now writes `_domain_label_extractor_missing_field_warning` tag to event tags in case of missing fields
* `geoip_enricher` now writes `_geoip_enricher_missing_field_warning` tag to event tags in case of missing fields
* `grokker` now writes `_grokker_missing_field_warning` tag to event tags instead of `_grokker_failure` in case of missing fields
* `requester` now writes `_requester_missing_field_warning` tag to event tags instead of `_requester_failure` in case of missing fields
* `timestamp_differ` now writes `_timestamp_differ_missing_field_warning` tag to event tags instead of `_timestamp_differ_failure` in case of missing fields
* `timestamper` now writes `_timestamper_missing_field_warning` tag to event tags instead of `_timestamper_failure` in case of missing fields
```

it is better scannable for breaking changes if we write the changed component first

ekneg54 commented on tests/unit/processor/template_replacer/test_template_replacer.py:
```diff
@@ -155,4 +159,9 @@ def test_replace_fails_with_invalid_template(self):
-            Factory.create({"test instance": config}, self.logger)
+            self._create_template_replacer(config)
+
+    def _create_template_replacer(self, config):
+        template_replacer = Factory.create({"test instance": config}, self.logger)
+        template_replacer.setup()
+        return template_replacer
```
consider to use a pytest fixture for this:


```suggestion
    @pytest.fixture(name="template_replacer")
    def _create_template_replacer(self, config):
        template_replacer = Factory.create({"test instance": config}, self.logger)
        template_replacer.setup()
        return template_replacer
```

and use it in the test methods parameter under its name.

ppcad replied:
I'm not sure how to use this, since it requires a parameter from the test.

I could do something like this:
```python
@pytest.fixture(name="template_replacer")
def fixture_create_template_replacer(self):
    def _create_template_replacer(config):
        template_replacer = Factory.create({"test instance": config}, self.logger)
        template_replacer.setup()
        return template_replacer
    return _create_template_replacer
```

ekneg54 reviewed (changes requested):
hello @ppcad,

I reviewed the PR. At first thank you for your commitment. This will make logprep a lot more concise and enables further refactorings and changes in architecture. Thank you for that.

Please have a look at my comments and suggestions.

ekneg54 commented:
hello @ppcad ,

I merged main and fixed some occuring errors. please have a look in the last two commits, that I do not accidentally broke something

dtrai2 commented on tests/unit/util/test_rule_dry_runner.py:
```diff
@@ -156,7 +156,6 @@ def test_dry_run_print_custom_output(self, tmp_path, capsys):
         dry_runner.run()
 
         captured = capsys.readouterr()
-        assert captured.err == ""
```
Why is this assertion being dropped?

ppcad replied:
It now creates a `"ProcessingWarning: pseudonymizer: missing source_fields`, since the pseudonymizer. So I removed it here.

dtrai2 reviewed (commented):
I only looked quickly over it without going much into detail. Two addition things I did notice though:

ppcad commented on logprep/abc/processor.py:
```diff
@@ -151,9 +154,11 @@ def process(self, event: dict):
            A dictionary representing a log event.
 
         """
+        self._extra_data = []
```
This needs to be cleared on the beggining of processing.
Maybe I could change this to self._extra_data.clear() to make this clear.

ppcad commented on logprep/processor/generic_resolver/processor.py:
```diff
@@ -78,47 +81,54 @@ def _apply_rules(self, event, rule):
+        target: str,
+        content: Union[str, float, int, list, dict],
+    ) -> bool:
+        """Extend list if content is not already in the list"""
```
I don't think it can be solved with a set, since list values could already exist in the event.
This could lead to duplicates.
A set would only ensure that newly added values are unique in regard to each other.

ppcad commented:
> hello @ppcad ,
> 
> I merged main and fixed some occuring errors. please have a look in the last two commits, that I do not accidentally broke something

Thanks, it looks fine to me.

> hello @ppcad ,
> 
> I merged main and fixed some occuring errors. please have a look in the last two commits, that I do not accidentally broke something

Thanks, it looks good to me.

ekneg54 commented:
Hello @ppcad ,

It Looks good to me.
Please feel free to merge it if you are ready.

ppcad commented:
I just fixed a problem with the clusterer.
Can you please have a look?
